packages/react/src/components/Checkbox/Checkbox.tsx

```diff
@@ -63,116 +78,146 @@ export interface CheckboxProps extends UIComponentProps, ChildrenComponentProps
+  const { setStart, setEnd } = useTelemetry(Checkbox.displayName, context.telemetry)
+  setStart()
+
+  const {
+    checked,
+    className,
+    defaultChecked,
+    design,
+    disabled,
+    label,
+    labelPosition,
+    icon,
+    styles,
+    toggle,
+    variables,
+  } = props
+
+  const { state, actions } = useStateManager(createCheckboxManager, {
+    mapPropsToInitialState: () => ({ checked: defaultChecked }),
+    mapPropsToState: () => ({ checked }),
+  })
+  const getA11Props = useAccessibility(props.accessibility, {
+    debugName: Checkbox.displayName,
+    mapPropsToBehavior: () => ({
+      checked: state.checked,
+      disabled,
     }),
-    checked: PropTypes.bool,
-    defaultChecked: PropTypes.bool,
-    disabled: PropTypes.bool,
-    icon: customPropTypes.itemShorthandWithoutJSX,
-    label: customPropTypes.itemShorthand,
-    labelPosition: PropTypes.oneOf(['start', 'end']),
-    onChange: PropTypes.func,
-    onClick: PropTypes.func,
-    toggle: PropTypes.bool,
-  }
-
-  static defaultProps = {
-    accessibility: checkboxBehavior,
-    icon: {},
-    labelPosition: 'end',
-  }
-
-  static autoControlledProps = ['checked']
-
-  actionHandlers = {
-    performClick: (e: any /* TODO: use React.KeyboardEvent */) => {
-      e.preventDefault()
-      this.handleClick(e)
+    actionHandlers: {
+      performClick: (e: React.KeyboardEvent) => {
+        e.preventDefault()
+        handleClick(e)
+      },
     },
-  }
+    rtl: context.rtl,
+  })
+  const { classes, styles: resolvedStyles } = useStyles(Checkbox.displayName, {
+    className: Checkbox.className,
+    mapPropsToStyles: () => ({
+      checked: state.checked,
+      disabled,
+      labelPosition,
+      toggle,
+    }),
+    mapPropsToInlineStyles: () => ({
+      className,
+      design,
+      styles,
+      variables,
+    }),
+    rtl: context.rtl,
+  })
 
-  getInitialAutoControlledState(): CheckboxState {
-    return { checked: false }
-  }
+  const ElementType = getElementType(props)
+  const unhandledProps = getUnhandledProps(Checkbox.handledProps, props)
 
-  handleChange = (e: React.ChangeEvent) => {
+  const handleChange = (e: React.ChangeEvent) => {
     // Checkbox component doesn't present any `input` component in markup, however all of our
     // components should handle events transparently.
-    const { disabled } = this.props
-    const checked = !this.state.checked
+    const checked = !state.checked
 
     if (!disabled) {
-      this.setState({ checked })
-      _.invoke(this.props, 'onChange', e, { ...this.props, checked })
+      actions.toggle(checked)
+      _.invoke(props, 'onChange', e, { ...props, checked })
     }
   }
 
-  handleClick = (e: React.MouseEvent | React.KeyboardEvent) => {
-    const { disabled } = this.props
-    const checked = !this.state.checked
+  const handleClick = (e: React.MouseEvent | React.KeyboardEvent) => {
+    const checked = !state.checked
```

Comment:
nit: move this line in the condition below

Reply:
Done 👍